src/encoded/commands/update_submitted_id_pattern.py

```diff
@@ -0,0 +1,322 @@
+    new_pattern = SubmittedIdPattern(
+        submission_center_code_pattern or parsed_existing_pattern.center_code,
+        parsed_existing_pattern.item_type,
+        identifier_pattern or parsed_existing_pattern.identifier,
+    )
+    return new_pattern.to_string()
+
+
+def add_new_pattern(
+    schema: Dict[str, Any],
+    schema_path: Path,
+    submission_center_code_pattern: str,
+    identifier_pattern: str,
+) -> None:
+    """Create pattern for item's schema, add to the submitted_id property,
+    and write the updated schema to file.
+    """
+    item_type = get_pattern_item_type(schema_path)
+    submitted_id_pattern = SubmittedIdPattern(
+        submission_center_code_pattern, item_type, identifier_pattern
+    ).to_string()
+    logger.info(
+        f"Adding submitted_id pattern {submitted_id_pattern} for schema"
+        f" {schema_path.name}"
+    )
+    update_schema_and_write_file(schema, submitted_id_pattern, schema_path)
+
+
+def get_pattern_item_type(schema_path: Path) -> str:
+    """Infer item type from schema file name."""
+    return schema_path.stem.upper().replace("_", "-")
+
+
+def update_schema_and_write_file(
+    schema: Dict[str, Any], submitted_id_pattern: str, schema_path: Path
+) -> None:
+    """Create schema with pattern and write content to file.
+
+    Since dealing with raw schema, not merged schema, check to see if
+    `submitted_id` present as property to overwrite mixin; if so, add to that,
+    otherwise create override.
+    """
+    submitted_id_property = get_submitted_id_property_with_pattern(
+        schema, submitted_id_pattern
+    )
+    update_schema(schema, submitted_id_property)
+    write_schema_file(schema_path, schema)
+
+
+def get_submitted_id_property_with_pattern(
+    schema: Dict[str, Any], submitted_id_pattern: str
+) -> Dict[str, Any]:
+    """Update or create `submitted_id` property for schema."""
+    existing_property = schema_utils.get_property(schema, SUBMITTED_ID_PROPERTY)
+    return {
+        **existing_property,
+        schema_utils.SchemaConstants.PATTERN: submitted_id_pattern
+    }
+
+
+def update_schema(
+    schema: Dict[str, Any], submitted_id_property: Dict[str, Any]
+) -> None:
+    """Overwrite or add `submitted_id` with pattern to raw schema."""
+    schema[schema_utils.SchemaConstants.PROPERTIES][
+        SUBMITTED_ID_PROPERTY
+    ] = submitted_id_property
+
+
+def load_raw_schema(schema_path: Path) -> Dict[str, Any]:
+    """Load schema is it's written (i.e. without resolving refs)."""
+    with schema_path.open() as file_handle:
+        return json.load(file_handle)
+
+
+def write_schema_file(
+    schema_path: Path, schema_contents: Dict[str, Any]
+) -> None:
+    """Write schema content to path."""
+    with schema_path.open("w") as file_handle:
+        json.dump(schema_contents, file_handle, indent=4)
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Update `submitted_id` pattern in all schemas"
```

Comment:
I was also curious about this.  Will this not potentially lead to a mass of validation errors - I don't know the status of upgraders in SMaHT but depending on the use case there would need to be some coupling there if changes are made after items have been submitted?

Reply:
Mostly, I decided to automate it because I didn't want to do the manual work, and if things change then the update will be simpler in the future.

Changes could cause validation errors, but that's not necessarily the case; depends on if the changes are backwards compatible or not.